### ISSUE TITLE
requestlog.py starts a thread without daemon=True

### ISSUE THREAD

**Issue by amckinley** [boto/boto#2723]
requestlog.py creates a thread and then throws away the reference to it, which will leave your program hanging on exit for that thread to exit.

**Comment by dkavanagh** [comment]
agreed and this is bad. It caused me issues when developing for a while.
